packages/cli/src/cli/cogify/action.cog.ts

```diff
@@ -126,6 +128,20 @@ export class ActionCogCreate extends CommandLineAction {
       } else {
         logger.warn('DryRun:Done');
       }
+    } catch (e) {
+      const jobId = Config.ProcessingJob.id(job.id);
+      const jobConfig = await Config.ProcessingJob.get(jobId);
+      if (jobConfig != null) {
+        const jobFailed = jobConfig as ProcessingJobFailed;
+        jobFailed.status = 'failed';
+        if (typeof e === 'string') {
+          jobFailed.error = e;
+        } else if (e instanceof Error) {
+          jobFailed.error = e.message; // works, `e` narrowed to Error
+        }
+        const config = new ConfigProviderDynamo(Const.TileMetadata.TableName);
```

Comment:
Why are you making a new config here? you already have `await Config.ProcessingJoob.get(...)`

Reply:
I need to use `config.ProcessingJob.put()`, we only have get function in `Config.ProcessingJob`

Reply:
I'm not sure I follow?

`config.ProcessingJob.put` is a function and it exists?

Reply:
Sorry, I need a put function but it is not exists in the `Config.ProcessingJob`, this `Config` is import from `@basemaps/shared` as the following code `export const Config = new ConfigInstance();`. 

To be able to put item into dynamodb, I need a `const config = new ConfigProviderDynamo(Const.TileMetadata.TableName);`

Reply:
You could use a `instanceof` check to see if it has a `.put()` like we do in basemaps-config?

https://github.com/linz/basemaps-config/blob/master/src/config/tileset.updater.ts#L122

Reply:
Let me try. Thanks.

Reply:
Works for me. Just updated.